coderd/database/dbtestutil/db.go

```diff
@@ -110,3 +126,75 @@ func dbNameFromConnectionURL(t testing.TB, connectionURL string) string {
 	require.NoError(t, err)
 	return strings.TrimPrefix(u.Path, "/")
 }
+
+func DumpOnFailure(t testing.TB, connectionURL string) {
+	if !t.Failed() {
+		return
+	}
+	cwd, err := filepath.Abs(".")
+	if err != nil {
+		t.Errorf("dump on failure: cannot determine current working directory")
+		return
+	}
+	snakeCaseName := regexp.MustCompile("[^a-zA-Z0-9-_]+").ReplaceAllString(t.Name(), "_")
+	outPath := filepath.Join(cwd, snakeCaseName+".test.sql")
```

Comment:
How about adding a random suffix to make sure we don't overwrite previous dumps?

Reply:
Could add `YYYYmmddHHMMSS` so you know at a glance which one is the most recent?